src/q6/calcu-INGI1123/mcq/QCM11.tex

```diff
@@ -8,15 +8,15 @@
   \mcq{Un problème qui peut être résolu par un algorithme (de complexité) polynomial est pratiquement faisable.}{1}
   {Pratiquement faisable = Polynomial}
   \mcq{Un problème qui peut être résolu par un algorithme (de complexité) exponentiel est pratiquement infaisable}{0}
-  {Un problème est intrinsèquement complexe, un algorithme est pratiquement infaisable}
+  {Il peut très bien être réalisable à la fois par un algorithme exponentiel ET un algorithme polynomial.}
   \mcq{Si un problème est intrinsèquement complexe en MT alors il est aussi intrinsèquement complexe pour le langage Java.}{1}
-  {}
+  {Le choix du modèle de calculabilité impacte la complexité temporelle. Néanmoins, la différence de complexité est tout au plus polynomiale entre tous les modèles de calcul déterministes. Or, la composition de fonctions polynomiales reste polynomiale!}
   \mcq{Un ensemble est a-réductible (algorithmiquement réductible) à son complément.}{1}
-  {}
-  \mcq{Tout ensemble récursivement énumérable est a-réductible à HALT.}{1}
+  {Dire que A est a-réductible à B signifie que l'algorithme pouvant décider B nous permet de décider A. Ici, décider le complément permet bien évidemment de décider l'ensemble initial, il suffit d'inverser la réponse.}
+  \mcq{Tout ensemble récursivement énumérable est a-réductible à HALT.}{0}
```

Comment:
Je pense justement que c'est vrai.
Pour un ensemble récursivement énumérable A, il existe un programme P qui renvoit 1 si x est dans A. Pour décider si x \in A, on demande à HALT si P termine avec l'input x. Si oui, alors on renvoit P(x), sinon on renvoit que x appartient pas à A.
Si HALT est récursif, A est donc aussi récursif